src/Sarif.Viewer.VisualStudio.Core/Models/SarifErrorListItem.cs

```diff
@@ -645,6 +642,42 @@ internal void PopulateAdditionalPropertiesIfNot()
             }
         }
 
+        internal void BuildRelatedLocationsTree()
+        {
+            var rootNode = new LocationModel();
+            LocationModel lastNode = rootNode;
+            int lastLevel = -1;
+
+            foreach (Location location in this.SarifResult.RelatedLocations)
+            {
+                var locationModel = location.ToLocationModel(this.SarifResult.Run, resultId: this.ResultId, runIndex: this.RunIndex);
+                int levelChange = locationModel.NestingLevel - lastLevel;
```

Comment:
>NestingLevel [](http://example.com/codeflow?start=48&length=12)

do we assume there always a first location in the array nesting level starts with 0? tested if first location nesting level is not 0, the tree doesn't render.

Reply:
Like I said in the description, I'm assuming the locations are numbered correctly. I'll ask Hwi-sung if this is safe to do.

Reply:
Confirmed that Kyle is making the same assumption